tests/lib/rules/prefer-named-capture-group.js

```diff
@@ -104,15 +124,61 @@ ruleTester.run("prefer-named-capture-group", rule, {
                 data: { group: "([0-9]{4})" },
                 line: 1,
                 column: 1,
-                endColumn: 21
+                endColumn: 21,
+                suggestions: [
+                    {
+                        messageId: "addGroupName",
+                        output: "RegExp('(?<temp1>[0-9]{4})')"
+                    },
+                    {
+                        messageId: "addNonCapture",
+                        output: "RegExp('(?:[0-9]{4})')"
+                    }
+                ]
             }]
         },
         {
             code: "new RegExp(`a(bc)d`)",
             errors: [{
                 messageId: "required",
                 type: "NewExpression",
-                data: { group: "(bc)" }
+                data: { group: "(bc)" },
+                suggestions: [
+                    {
+                        messageId: "addGroupName",
+                        output: "new RegExp(`a(?<temp1>bc)d`)"
+                    },
+                    {
+                        messageId: "addNonCapture",
+                        output: "new RegExp(`a(?:bc)d`)"
+                    }
+                ]
+            }]
+        },
+        {
+            code: "new RegExp('\u1234\u5678(?:a)(b)');",
+            errors: [{
+                messageId: "required",
+                type: "NewExpression",
+                data: { group: "(b)" },
+                suggestions: [
+                    {
+                        messageId: "addGroupName",
+                        output: "new RegExp('\u1234\u5678(?:a)(?<temp1>b)');"
+                    },
+                    {
+                        messageId: "addNonCapture",
+                        output: "new RegExp('\u1234\u5678(?:a)(?:b)');"
+                    }
+                ]
+            }]
+        },
+        {
+            code: "new RegExp('\\u1234\\u5678(?:a)(b)');",
+            errors: [{
+                messageId: "required",
+                type: "NewExpression",
+                data: { group: "(b)" }
             }]
         },
```

Comment:
Can you also add `suggestions: null` in test cases where we're expecting no suggestions so that all tests have a `suggestions` property? Omitting ` suggestions` does not assert that there were no suggestions provided, it just doesn't check anything related to suggestions. This will be improved per [RFC #84](https://github.com/eslint/rfcs/tree/main/designs/2021-stricter-rule-test-validation) in the next major version, but at the moment we have to add `suggestions: null` if we want to check that the rule didn't provide suggestions.